xbmc/input/ButtonStat.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (C) 2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+#include "ButtonStat.h"
+
+#include "InputTypes.h"
+
+using namespace KODI;
+using namespace INPUT;
+
+CButtonStat::CButtonStat() = default;
+CButtonStat::~CButtonStat() = default;
+
+CKey CButtonStat::TranslateKey(CKey key) const
+{
+  uint32_t buttonCode = key.GetButtonCode();
+  if (key.GetHeld() > HOLD_TRESHOLD)
+    buttonCode |= CKey::MODIFIER_LONG;
+
+  CKey translatedKey(buttonCode, key.GetHeld());
```

Comment:
Can we not do `key.SetButtonCode();`? Is it ok to drop non-code or held CKey properties?

Reply:
Nevermind, this class is called `CButtonState`, and buttons are scoped to only have a code and held duration.